File: leetcode/0217-ContainsDublicate/solution.cpp
// Compute complexity = O(N)
// Space complexity = O(N)
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        std::unordered_set<int> occurances;
        for (const auto& n : nums) {
            if (occurances.end() != occurances.find(n)) {
                return true;
            }
            occurances.insert(n);
        }
        return false;
    }
};

File: leetcode/0217-ContainsDublicate/solution.py
# Compute complexity = O(N)
# Space complexity = O(N)
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        occurances = set()
        for n in nums:
            if n in occurances:
                return True
            occurances.add(n)
        return False
